src/libs/Navigation/helpers/navigateToWorkspacesPage.ts

```diff
@@ -0,0 +1,79 @@
+import interceptAnonymousUser from '@libs/interceptAnonymousUser';
+import {getPreservedNavigatorState} from '@libs/Navigation/AppNavigator/createSplitNavigator/usePreserveNavigatorState';
+import Navigation from '@libs/Navigation/Navigation';
+import navigationRef from '@libs/Navigation/navigationRef';
+import {isPendingDeletePolicy, shouldShowPolicy as shouldShowPolicyUtil} from '@libs/PolicyUtils';
+import CONST from '@src/CONST';
+import NAVIGATORS from '@src/NAVIGATORS';
+import ROUTES from '@src/ROUTES';
+import SCREENS from '@src/SCREENS';
+import type {Policy} from '@src/types/onyx';
+import {isFullScreenName, isWorkspacesTabScreenName} from './isNavigatorName';
+import {getLastVisitedWorkspaceTabScreen, getWorkspacesTabStateFromSessionStorage} from './lastVisitedTabPathUtils';
+
+type Params = {
+    currentUserLogin?: string;
+    shouldUseNarrowLayout: boolean;
+    policy?: Policy;
+};
+
+const getWorkspaceNavigationRouteState = () => {
+    const rootState = navigationRef.getRootState();
+    const topmostFullScreenRoute = rootState?.routes?.findLast((route) => isFullScreenName(route.name));
+    if (!topmostFullScreenRoute) {
+        return {};
+    }
+    const workspacesTabStateFromSessionStorage = getWorkspacesTabStateFromSessionStorage() ?? rootState;
+    const lastWorkspacesTabNavigatorRoute = workspacesTabStateFromSessionStorage?.routes.findLast((route) => isWorkspacesTabScreenName(route.name));
+    let workspacesTabState = lastWorkspacesTabNavigatorRoute?.state;
+
+    if (!workspacesTabState && lastWorkspacesTabNavigatorRoute?.key) {
+        workspacesTabState = getPreservedNavigatorState(lastWorkspacesTabNavigatorRoute.key);
+    }
+
+    return {lastWorkspacesTabNavigatorRoute, workspacesTabState, topmostFullScreenRoute};
+};
+
+const navigateToWorkspacesPage = ({currentUserLogin, shouldUseNarrowLayout, policy}: Params) => {
+    const {lastWorkspacesTabNavigatorRoute, topmostFullScreenRoute} = getWorkspaceNavigationRouteState();
+    if (!topmostFullScreenRoute) {
```

Comment:
Can you please add some comments to the code that clearly explains _WHY_ each of these alternate navigation paths are necessary?